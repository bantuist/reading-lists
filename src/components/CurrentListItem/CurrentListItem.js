import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

function CurrentListItem({ book = [], onUpdateBook, onRemoveBook }) {
  const { imageLinks, title, authors, amount, id, isRead } = book;
  let checkBook = 'check-square-o';
  if (isRead) {
    checkBook = 'check-square';
  }

  return (
    <li className="Book-list-item" key={id}>
      <div className="image-container">
        <img src={imageLinks} alt="suggestion" />
      </div>
      <div className="Book-details">
        <h3 className="Book-title">{title}</h3>
        <p className="Book-author">{authors}</p>
        {typeof amount === 'number' &&
          <p className="Book-amount">${amount}</p>}
        <p className="Book-id">{id}</p>
      </div>
      <FontAwesome 
        name={checkBook}
        onClick={() => onUpdateBook(book)} 
      />
      <button 
        className="Book-remove" 
        onClick={() => onRemoveBook(id)}
        >X</button>
    </li>
  );
}

CurrentListItem.propTypes = {
  book: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    authors: PropTypes.array.isRequired,
    amount: PropTypes.oneOfType([
      PropTypes.number, 
      PropTypes.string]
    ),
    imageLinks: PropTypes.string.isRequired,
    isRead: PropTypes.bool.isRequired
  }),
}

export default CurrentListItem;