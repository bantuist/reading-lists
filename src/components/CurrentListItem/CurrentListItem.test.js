import React from 'react';
import ReactDOM from 'react-dom';
import CurrentListItem from './CurrentListItem';
import renderer from 'react-test-renderer';

it('should render CurrentListItem', () => {
  const tree = renderer.create(<CurrentListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});