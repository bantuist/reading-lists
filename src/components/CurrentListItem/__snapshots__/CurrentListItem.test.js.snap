// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentListItem 1`] = `
<li
  className="Book-list-item"
>
  <div
    className="image-container"
  >
    <img
      alt="suggestion"
      src={undefined}
    />
  </div>
  <div
    className="Book-details"
  >
    <h3
      className="Book-title"
    />
    <p
      className="Book-author"
    />
    <p
      className="Book-id"
    />
  </div>
  <span
    aria-hidden={true}
    className="fa fa-check-square-o"
    onClick={[Function]}
  />
  <button
    className="Book-remove"
    onClick={[Function]}
  >
    X
  </button>
</li>
`;
